src/include/daos/object.h

```diff
@@ -205,6 +205,7 @@ struct daos_obj_layout {
 struct daos_shard_tgt {
 	uint32_t		st_rank;	/* rank of the shard */
 	uint32_t		st_shard;	/* shard index */
+	uint32_t		st_shard_id;	/* shard id */
```

Comment:
maybe better to add a uint32_t padding to make it 8B aligned? as it is in RPC

Reply:
we share this structure with client daos task stack so if I add another padding here, then I will need shrink other stuff further

Reply:
it seems separate the structure might need extra allocation (or cycle) in the hot path, so probably just adding pad might be better here.

Reply:
You changed the RPC layout, so need bump the RPC version for interoperability.

Reply:
sure.